### ISSUE TITLE
config nu fails on windows "External command failed"

### ISSUE THREAD

**Issue by NogrTL** [nushell/nushell#6646]
### Describe the bug

While trying to run `config nu` received error
```
Error: nu::shell::external_command (link)

  × External command failed
   ╭─[source:1:1]
 1 │ nu
   · ▲
   · ╰── can't run executable
   ╰────
  help: program not found
```

### How to reproduce

1. Run `config nu`

### Expected behavior

Open config?

### Screenshots

![image](https://user-images.githubusercontent.com/12021449/193268599-63aeab5b-18ff-48ea-b592-876c3c920544.png)

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.69.1                                          |
| branch             |                                                 |
| commit_hash        | 8a5abc7afc28a36908a32f1852127e2a5170b461        |
| build_os           | windows-x86_64                                  |
| build_target       | x86_64-pc-windows-msvc                          |
| rust_version       | rustc 1.64.0 (a55dd71d5 2022-09-19)             |
| rust_channel       | stable-x86_64-pc-windows-msvc                   |
| cargo_version      | cargo 1.64.0 (387270bc7 2022-09-16)             |
| pkg_version        | 0.69.1                                          |
| build_time         | 2022-09-28 02:58:08 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

**Comment by sholderbach** [comment]
OK, can reproduce this under a certain condition. This is the result of it looking for a text editor specified either via the `config.buffer_editor` setting or the `EDITOR` or `VISUAL` environment variables and failing to locate that in the PATH.
The error message is absolutely misleading and should not highlight `nu` and instead report details about the editor we search for.

**Comment by fdncred** [comment]
maybe we should default to `notepad` on windows. I haven't looked but maybe it defaults to `nano` even on windows?

**Comment by sholderbach** [comment]
I think we do already, at least on the Rust side. Don't know, if our overly eager default config overwrites this.

**Comment by NogrTL** [comment]
I know the answer...In new version's of Windows 10, you can remove it. I guess that's the problem

**Comment by fdncred** [comment]
oh ya, good point @NogrTL. Is there another editor that we can rely on to be on windows?

**Comment by NogrTL** [comment]
I would suggest, to dig into handlers if file-types and protocols handlers.

It's more of a GUI approach...But it cat fall back to select any program you want to open file

**Comment by fdncred** [comment]
Maybe we should just check if notepad exists on windows and fail with an error message that talks about setting the proper env var. I'm not personally interested in jumping though windows file association hoops in the hope to find something associated with `.nu`. 

I just wonder if looking up what is associated with `.txt` is worth the hassle. I may accept a PR to do such a thing though as settings those env vars is not a normal Windows thing.

**Comment by sholderbach** [comment]
Error message fixed by #6730
